src/backend/message-port-like.js

```diff
@@ -0,0 +1,76 @@
+import {TypedEmitter} from 'tiny-typed-emitter';
+
+/**
+ * @typedef {Object} MessagePortEvents
+ * @property {(value: any) => void} message
+ */
+
+/** @typedef {import('nodejs-mobile-react-native').Channel} Channel */
+
+/**
+ * Wrap the nodejs-mobile RNBridge to act like a MessagePort. Queues messages
+ * until `start()` is called.
+ * @extends {TypedEmitter<MessagePortEvents>}
+ */
+class MessagePortLike extends TypedEmitter {
+  /** @type {any[]} */
+  #queuedMessages = [];
+  /** @type {'idle' | 'started' | 'closed'} */
+  #state = 'idle';
+  /** @type {(message: any) => void} */
+  #messageHandler;
+  /** @type {String} */
+  #API_EVENT_NAME = '@@API_MESSAGE';
+  /** @type {Channel} */
+  #channel;
+
+  /**
+   * @param {Channel} channel
+   */
+  constructor(channel) {
```

Comment:
Since this is strictly tied to the static import `nodejs-mobile-react-native`.Channel, I think this should just be an import in this file, and not a constructor parameter, so this file encapsulates the Channel dependency.